pg_search/src/api/operator.rs

```diff
@@ -206,42 +185,39 @@ unsafe fn make_search_config_opexpr_node(
     (*var).vartypmod = att.atttypmod;
     (*var).varcollid = att.attcollation;
 
-    let have_query = query.is_some();
-    if let Some(query) = query {
-        // fabricate a `SearchConfig` from the above relation and query string
-        // and get it serialized into a JSONB Datum
-        let search_config = SearchConfig::from((query, &indexrel));
-
-        let search_config_json =
-            serde_json::to_value(&search_config).expect("SearchConfig should serialize to json");
-        let jsonb_datum = JsonB(search_config_json).into_datum().unwrap();
+    // we're about to fabricate a new pg_sys::OpExpr node to return
+    // that represents the `@@@(anyelement, jsonb)` operator
+    let mut newopexpr = PgBox::<pg_sys::OpExpr>::alloc_node(pg_sys::NodeTag::T_OpExpr);
 
-        // from which we'll create a new pg_sys::Const node
-        let jsonb_const = pg_sys::makeConst(
-            pg_sys::JSONBOID,
+    if let Some(query) = query {
+        // In case a sequential scan gets triggered, we need a way to pass the index oid
+        // to the scan function. It otherwise will not know which index to use.
+        let wrapped_query = SearchQueryInput::WithIndex {
+            oid: indexrel.oid(),
+            query: Box::new(query),
+        };
+
+        // create a new pg_sys::Const node
+        let search_query_input_const = pg_sys::makeConst(
+            pgrx::pg_sys::Oid::from(16735), // SearchQueryInput::type_oid(),
```

Comment:
Oh no!  Can't hardcode a random Oid!

```rust
pub fn fieldname_typoid() -> pg_sys::Oid {
    unsafe {
        let oid = direct_function_call::<pg_sys::Oid>(
            pg_sys::regtypein,
            &[c"paradedb.SearchQueryInput".into_datum()],
        )
        .expect("type `paradedb.SearchQueryInput` should exist");
        if oid == pg_sys::Oid::INVALID {
            panic!("type `paradedb.SearchQueryInput` should exist");
        }
        oid
    }
}
```

Put that somewhere and call it to get the Oid.